src/test/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectDataForReadOrientationFilterIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+
+        /** Expected result
+         *
+         * (position chr1: 100,000,000-100,000,011)
+         *
+         * ref: T C A C T A A G C A C A
+         * alt: x C T C T G A C C A A x
+         *          *     *   *     *
+         *
+         * At each alt site, we have 75 ref and 25 alt reads
+         *
+         * context, alt counts, alt f1r2 counts, depth
+         * -------------------------------------------
+         * CAC, {70, 0, 0, 30}, {35, 0, 0, 15}, 100
+         * TAA, {70, 0, 30, 0}, {35, 0, 15, 0}, 100
+         * AGC, {0, 30, 70, 0}, {0, 15, 35, 0}, 100
+         * ACA, {30, 70, 0, 0}, {15, 35, 0, 0}, 100
+         *
+         *
+         * Ref: 6 sites at depth 100
+         * TCA, ACT, CTA, AAG, GCA, CAC
+         **/
+
+        // alt site tests
+        Assert.assertEquals(altDesignMatrix.size(), 4);
+        AltSiteRecord recordCAC = altDesignMatrix.stream().
+                filter(rec -> rec.getReferenceContext().equals("CAC")).findFirst().get();
+        ArrayAsserts.assertArrayEquals(recordCAC.getBaseCounts(), new int[]{70, 0, 0, 30});
+        ArrayAsserts.assertArrayEquals(recordCAC.getF1R2Counts(), new int[]{35, 0, 0, 15});
+
+        AltSiteRecord recordTAA = altDesignMatrix.stream().
+                filter(rec -> rec.getReferenceContext().equals("TAA")).findFirst().get();
+        ArrayAsserts.assertArrayEquals(recordTAA.getBaseCounts(), new int[]{70, 0, 30, 0});
+        ArrayAsserts.assertArrayEquals(recordTAA.getF1R2Counts(), new int[]{35, 0, 15, 0});
+
+        AltSiteRecord recordAGC = altDesignMatrix.stream().
+                filter(rec -> rec.getReferenceContext().equals("AGC")).findFirst().get();
+        ArrayAsserts.assertArrayEquals(recordAGC.getBaseCounts(), new int[]{0, 30, 70, 0});
+        ArrayAsserts.assertArrayEquals(recordAGC.getF1R2Counts(), new int[]{0, 15, 35, 0});
+
+        AltSiteRecord recordACA = altDesignMatrix.stream().
+                filter(rec -> rec.getReferenceContext().equals("ACA")).findFirst().get();
+        ArrayAsserts.assertArrayEquals(recordACA.getBaseCounts(), new int[]{30, 70, 0, 0});
+        ArrayAsserts.assertArrayEquals(recordACA.getF1R2Counts(), new int[]{15, 35, 0, 0});
+
+        // check ref histograms
+        for (String exptectedRefContext : Arrays.asList("TCA", "ACT", "CTA", "AAG", "GCA", "CAC")){
+            RefSiteHistogram histogram = histograms.stream()
+                    .filter(hist -> hist.getReferenceContext().equals(exptectedRefContext))
+                    .findFirst().get();
+            // recall the refsite histogram places depth 1 at index 0
+            // index: 0, 1, ..., depth-1, ...
+            // depth: 1, 2, ..., depth,   ...
+            Assert.assertEquals(histogram.getCounts()[depth-1], 1);
+            Assert.assertEquals(MathUtils.sum(histogram.getCounts()), 1);
+        }
+    }
+
+    private File createSyntheticSam(final int numRefReads, final int numAltReads) throws IOException {
```

Comment:
Just making sure you've checked that this doesn't duplicate any utility method that already exists.

Reply:
I did look around for a utility method. This comes up often enough too, but each time the characteristics of the artificial sam are different enough (e.g. in one case you want reads to have strand artifact, in another you want read orientation artifact, etc.) such that you have to make a sam by hand for each scenario, as I did here. It's a huge pain.